pennylane/transforms/core/transform_program.py

```diff
@@ -130,6 +130,12 @@ def __getitem__(self, idx):
     def __bool__(self):
         return bool(self._transform_program)
 
+    def __add__(self, other):
+        program = TransformProgram(self._transform_program)
+        for container in other:
+            program.push_back(container)
```

Comment:
since these have already been validated, can you just do `return TransformProgram(self._transform_program + other._transform_program)`?

Reply:
You cannot do it because, two valid program do not necessarily represent a valid program together

Reply:
Why?